maat-court-data-api/src/main/java/gov/uk/courtdata/controller/AfterResponseClearDown.java

```diff
@@ -0,0 +1,33 @@
+package gov.uk.courtdata.controller;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.annotation.WebFilter;
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@WebFilter("/*")
+@Component
+@Order(Ordered.LOWEST_PRECEDENCE)
+public class AfterResponseClearDown implements Filter {
+    
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+        ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        try {
+            filterChain.doFilter(servletRequest, servletResponse);
+        } finally {
+            log.debug("About to clear down the MDC");
+            MDC.clear();
+        }
+    }
+}
```

Comment:
This filter runs last during the request processing and first during the response processing. If any of the other filters uses the MDC in their finally blocks it's not going to log properly _I think_.